static/js/components/EducationTab.js

```diff
@@ -0,0 +1,240 @@
+    return requiredFields.concat(
+      ...profile.education.map((v, i) => nestedFields(i))
+    );
+  }
+
+  openNewEducationForm(level, index) {
+    const {dispatch, profile, updateProfile} = this.props;
+    let new_index = index;
+    if (index === null){
+      new_index = profile['education'].length;
+    }
+    /* add empty education */
+    let clone = Object.assign({}, profile);
+    clone['education'] = clone['education'].concat(generateNewEducation(level));
+    updateProfile(clone);
+    dispatch(openEducationForm(level, new_index));
+  }
+
+  printAddDegree(level){
+    const { educationLevels } = this.props;
+    if (educationLevels[level.value]){
+      return <Grid key={"add-"+level.value}>
+        <Cell col={11}></Cell>
+        <Cell col={1}>
+          <FABButton mini onClick={this.openNewEducationForm.bind(this, level.value, null )} raised ripple>
+            <Icon name="add" />
+          </FABButton>
+        </Cell>
+      </Grid>;
+    }
+    return null;
+  }
+
+  printExistingEducations(level) {
+    const {profile} = this.props;
+    return profile['education'].map(education => {
+      if (education.degree_name === level.value && 'id' in education) {
+        return <Grid className="existing-education-grid">
+          <Cell col={3}>{this.educationLevelLabels[education.degree_name]}</Cell>
+          <Cell col={7}>{education.graduation_date}</Cell>
+          <Cell col={1}>
+            <Button onClick={this.openEditEducationForm.bind(this, education.degree_name, education.id)}>
+              <Icon name="edit"/>
+            </Button>
+          </Cell>
+          <Cell col={1}>
+            <Button ><Icon name="delete"/></Button>
+          </Cell>
+        </Grid>;
+      }
+    });
+  }
+
+  openEditEducationForm(level, education_id){
+    const { dispatch, profile } = this.props;
+
+    let index = profile['education'].findIndex((education) => {
+      return ('id' in education) && education_id === education.id;
+    });
+    dispatch(openEducationForm(level, index));
+  }
+
+  saveEducationForm(nestedValidationCallback){
+    const { dispatch, saveProfile, profile, requiredFields, validationMessages } = this.props;
+    let fields;
+    if (_.isFunction(nestedValidationCallback)) {
+      fields = nestedValidationCallback(profile, requiredFields);
+    } else {
+      fields = requiredFields;
+    }
+    saveProfile(profile, fields, validationMessages ).then(() => {
+      dispatch(closeEducationForm());
+    });
+  }
+
+  handleSwitchClick(level){
+    const {dispatch, educationLevels} = this.props;
+    let newState = Object.assign({}, educationLevels);
+    newState[level] = !educationLevels[level];
+    dispatch(toggleEducationLevel(newState));
+  }
+
+  handleCloseDialog(){
+    const { dispatch, updateProfile, profile, educationDialog} = this.props;
+    dispatch(closeEducationForm());
+    let clone = Object.assign({}, profile);
+    clone['education'].splice(educationDialog.educationIndex, 1);
```

Comment:
We can't use mutable operations on objects that come from `props` since React requires that we treat them as immutable. `Object.assign` only does a shallow clone above which means `clone['education']` is still the same object as `profile['education']'. You will need to make a copy of the list first or use `_.cloneDeep` above. Although see my previous comment about reverting `profile.edit` entirely.